schedulearn-frontend/src/components/Suggestions/SuggestionForm.tsx

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import "./SuggestionForm.scss";
+import TopicList, { TopicListItem } from "../../server-components/TopicList";
+import { NewSuggestion } from "./CreateSuggestion";
+
+export interface SuggestionFormProps {
+  isOpen: boolean;
+  suggestion?: NewSuggestion;
+  submitText?: string;
+  onEventSubmit: (topicId: number) => void;
+}
+
+export interface SuggestionFormState {
+  topicId: number;
+  suggesteeId: number;
+}
+
+export class SuggestionForm extends React.Component<SuggestionFormProps, SuggestionFormState> {
+  public constructor(props: SuggestionFormProps) {
+    super(props);
+    this.state = this.getDefaultState();
+  }
+
+  private getDefaultState(): SuggestionFormState {
+    return {
+      topicId: this.props.suggestion?.topicId ?? 0,
+      suggesteeId: this.props.suggestion?.suggesteeId ?? 0,
+    };
+  }
+
+  handleSubmit = (event: React.MouseEvent<HTMLInputElement, MouseEvent>): void => {
+    event.preventDefault();
+    this.props.onEventSubmit(this.state.topicId);
+  }
+
+  componentDidUpdate(prevProps: SuggestionFormProps): void {
+    if ((this.props.isOpen !== prevProps.isOpen && this.props.isOpen === true) ||
+      this.props.suggestion?.suggesteeId !== prevProps.suggestion?.suggesteeId ||
+      this.props.suggestion?.topicId !== prevProps.suggestion?.topicId) {
+      this.setState(this.getDefaultState());
+    }
+  }
+
+  onTopicSelectChange = (topic: TopicListItem): void => {
+    if (topic.parentTopicId !== undefined)
+      this.setState({ topicId: topic.parentTopicId });
```

Comment:
This should not be parent id, rather it should be the topics own id, which can never be undefined.